src/lib/peer.js

```diff
@@ -59,12 +61,22 @@ function Peer(config) {
             self.fingerTable = new FingerTable(self.peerId, 
                                                self.events,
                                                self.channelManager);
+            var readyEventSent = false;
+            self.fingerTable.events.on('fingerUpdate', function(data){
+                if(readyEventSent) return;
+                //Send the ready event for the first 'fingerUpdate' event
+                readyEventSent = true;
+                self.events.emit('ready', {});
+            });
             self.events.emit('registered', {peerId: data.peerId});
         }
 
-        ioc.emit('s-register', {});
+        ioc.emit('s-register', {'id' : peerId});
```

Comment:
right now, the Signalling Server pick the ID's, are you proposing to change that?